kernel/Kconfig

```diff
@@ -600,6 +603,13 @@ config SYSTEM_WORKQUEUE_NO_YIELD
 	  cooperative and a sequence of work items is expected to complete
 	  without yielding.
 
+config SYSTEM_WORKQUEUE_WORK_TIMEOUT_MS
+	int "Select system work queue work timeout in milliseconds"
+	default 10000 if DEBUG
```

Comment:
CONFIG_DEBUG is a control over compiler optimizations, it's the wrong tunable to use here.  You probably want CONFIG_ASSERT to gate the default.

Reply:
Changed to ASSERT and decreased the time to 5000ms (still far longer than I think is reasonable but hoefully users will adjust it lower)